### PR TITLE
[Feature] CON 81 - Enable text wrap passages

### PR DESCRIPTION
**Pull request by habemuscode**
**Related to this issue**

https://oat-sa.atlassian.net/browse/CON-81

**Other related PRs**

- https://github.com/oat-sa/extension-tao-itemqti/pull/1580
- https://github.com/oat-sa/extension-tao-mediamanager/pull/240

**Story**

As the author, I want to be able to wrap my text around pictures, so that I can produce passages that look like real-life media content.

Currently, I am only able to anchor images AS a character. In order for the text to be able to flow around it, I will need to be able to anchor it TO a character and specify how text is supposed to wrap around it

Required wrap modes:

- Float off
- Float right
- Float left

**Acceptance Criteria**

- Anchoring an image as a character should still be possible
- Text wrapping should not be available when the image is anchored as a character
- Images maintain position within their paragraph when text is added before or after
- Responsive images are still supported if the image is anchored to a character
- Exported items will contain the relevant CSS to ensure the rendering will be correct in any environment
- Exporting and reimporting an item will maintain the anchoring and wrapping

### CODE REVIEW

File: src/mediaEditor/plugins/mediaAlignment/helper.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const $container = widget.$container,
+        elt = widget.element;
+
+    $container.removeClass('rgt lft');
+
+    elt.removeClass('rgt');
+    elt.removeClass('lft');
+    switch(position) {
+        case 'right':
+            $container.addClass('rgt');
```
Comment (bziondik) [inline src/mediaEditor/plugins/mediaAlignment/helper.js:30]: Check with new TR team is this classes will work for them

Comment (habemuscode) [reply src/mediaEditor/plugins/mediaAlignment/helper.js:30]: Done, needs to fix, nothing to do in our side

**Review by mike-ionut-mihai-sandu-tao — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Comment by mike-ionut-mihai-sandu-tao** [comment]
can we have a demo on this please

**Review by jsconan — COMMENTED** [review]
Unfortunately, I was unable to test on my env. I got several 500 errors.
Hence, this is a code review only.

Mainly my remarks are about the code style, especially about ES6 that could be used more.

The use of radio buttons looks a bit odd, and not really following the proper usage for [radio groups](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#Defining_a_radio_group).

Also, is it not possible to have it unit tested too?

**Review by gabrielfs7 — APPROVED** [review]
Code sounds good. Lets have a small demo.

Needs to resolve the conflicts first :) 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: src/mediaEditor/plugins/mediaAlignment/tpl/mediaAlignment.tpl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<h3 class="media-sizer-panel-label">{{__ 'Position'}}</h3>
+<div class="media-alignment">
```
Comment (404passionFruit) [inline src/mediaEditor/plugins/mediaAlignment/tpl/mediaAlignment.tpl:2]: Inputs can be grouped with fieldset tag, this is more semantical and better for a11y
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset

**Comment by 404passionFruit** [comment]
For some reason `src/mediaEditor/plugins/mediaAlignment/style.css.map` is empty

**Comment by 404passionFruit** [comment]
Converted to draft as there is an alternative solution https://github.com/oat-sa/tao-core-ui-fe/pull/302

**Review by habemuscode — COMMENTED** [review]
Also, the space between `Position` title and their group of things isnt the same like in `Size and position` title and their group

**Comment by habemuscode** [comment]
Closed in favor of https://github.com/oat-sa/tao-core-ui-fe/pull/302

**Pull request closed without merging**